### PR TITLE
Or 556/introduce lists

### PR DESCRIPTION
**Pull request by frshwtr**
## Describe your changes
image

image

## Checklist before requesting a review

- [ ] I have applied `percy` label for o-[COMPONENT] or `chromatic` label for o3-[COMPONENT] on my PR before merging and after review. Find more details in [CONTRIBUTING.md](https://github.com/Financial-Times/origami/blob/main/CONTRIBUTING.md#pull-requests-and-visual-regression-tests)
- [ ] If it is a new feature, I have added thorough tests.
- [ ] I have updated relevant docs.
- [ ] I have updated relevant env variables in Doppler.

### CODE REVIEW

File: components/o3-typography/list.css [hunk 1]
```diff
@@ -0,0 +1,43 @@
+.o3-typography-list {
+	margin: 0 0 12px;
```
Comment (frshwtr) [inline components/o3-typography/list.css:2]: Spacing tokens can be used here.

File: components/o3-typography/list.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+		}
+	}
+}
+
+.o3-typography-list--unordered {
+	> li {
+		&:before {
+			content: "\2022";
+			color: inherit;
+			transform: scale(1.5555555556);
```
Comment (frshwtr) [inline components/o3-typography/list.css:24]: Are these actual hard values or just generated? Double check legacy sass

File: components/o3-typography/stories/story-templates.tsx [hunk 3]
```diff
@@ -100,6 +97,46 @@ const LinkStory = {
...
+const UnorderedListStory = {
+	...TemplateSBConfig,
+	render: args => {
+		return (
+			<>
+				<Heading level="1" theme={args.theme}>
+					Fun Origami Facts
+				</Heading>
+				<Body theme={args.theme}>
+					<UnorderedList>
```
Comment (notlee) [inline components/o3-typography/stories/story-templates.tsx:109]: We can't include lists within a `p` element. It's broken the html tab in storybook
https://caninclude.glitch.me/caninclude?child=ul&parent=p

File: components/o3-typography/stories/story-templates.tsx [hunk 4]
```diff
@@ -100,6 +97,46 @@ const LinkStory = {
...
 };
 
+const UnorderedListStory = {
+	...TemplateSBConfig,
+	render: args => {
+		return (
+			<>
+				<Heading level="1" theme={args.theme}>
+					Fun Origami Facts
+				</Heading>
```
Comment (notlee) [inline components/o3-typography/stories/story-templates.tsx:107]: Do you think we could remove the headings to reduce the Story to what it is we are demonstrating?

Comment (notlee) [reply components/o3-typography/stories/story-templates.tsx:107]: Unrelated to this PR: It appears to be FinancierDisplay?

**Comment by frshwtr** [comment]
The readme needs updating to include this component.

File: components/o3-typography/list.css [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+.o3-typography-list--ordered {
+	counter-reset: item;
+	> li {
+		&:before {
+			font-weight: var(--o3-font-weight-semibold);
+			content: counter(item);
+			counter-increment: item;
+			font-feature-settings: "tnum";
+			transform: scale(0.8888888889);
```
Comment (ob6160) [inline components/o3-typography/list.css:39]: do we still need magic numbers like these?

File: components/o3-typography/README.md [hunk 6]
```diff
@@ -74,6 +75,39 @@ import { Link, Body } from '@financial-times/o3-typography';
...
+	<li>Item 1</li>
+	<li>Item 2</li>
+	<li>Item 3</li>
+</OrderedList>
+
+<UnorderedList>
+	<li>Item 1</li>
+	<li>Item 2</li>
+	<li>Item 3</li>
+</UnorderedList>
```
Comment (ob6160) [inline components/o3-typography/README.md:109]: tiny thing, maybe we could have these ordered (heh) the same way as the pure html examples above? (ul and then ol)

**Review by ob6160 — COMMENTED** [review]
small comments, otherwise great 💎 
can't wait to add this into my migration PR over to o3 typography for pro central banking 🚗

File: components/o3-typography/README.md (hunk 6 above)
Comment (akomiqaia) [reply components/o3-typography/README.md:109]: Presumable the second example is JSX template example. Maybe we should make that clear and say that second example would be how you would import it in JSX project.

File: components/o3-typography/list.css (hunk 5 above)
Comment (frshwtr) [reply components/o3-typography/list.css]: @ob6160 changed these to a calculation. In the old SASS these were computed.

I'd love to use custom properties here but it looks like the original calculation uses `px` values. Our new vars are `rem` :(

File: components/o3-typography/list.css [hunk 7]
```diff
@@ -0,0 +1,43 @@
+.o3-typography-list {
```
Comment (notlee) [inline components/o3-typography/list.css:1]: **Non blocking suggestion**: Is o3 an opportunity to simplify markup?

Something like:
```diff
+<ul class="o3-typography-ul"> [...] </ul>
-<ul class="o3-typography-list o3-typography-list--unordered"> [...] </ul>
```

We should continue to use BEM but I'm unconvinced it provides much benefit for simple, familiar elements.

File: components/o3-typography/list.css [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+	margin: 0 0 var(--o3-spacing-3xs);
+	padding: 0;
+	list-style: none;
+	> li {
+		position: relative;
+		padding-left: calc(2ch + 4px);
+
+		&:before {
```
[the remark this answers is not in the document]
Comment (notlee) [reply components/o3-typography/list.css:9]: Yes! All covered here https://docs.google.com/document/d/1RuGduWdX0zGsgsp9C7lIhXgqEia6sWK900_3XVwYDIM/edit#heading=h.1st2yski7d9i

**Review by notlee — APPROVED** [review]

File: components/o3-typography/src/tsx/list.tsx [hunk 9]
```diff
@@ -1,18 +1,10 @@
 type UnorderedListProps = React.HTMLAttributes<HTMLUListElement>;
 
 export const UnorderedList = ({children}: UnorderedListProps) => {
-	return (
-		<ul className="o3-typography-list o3-typography-list--unordered">
-			{children}
-		</ul>
-	);
+	return <ul className="o3-typography o3-typography-ul">{children}</ul>;
```
Comment (notlee) [inline components/o3-typography/src/tsx/list.tsx:4]: Ooops. I approved but do we need an `o3-typography` here and below?

**Pull request merged**